app/scripts/lib/multichain-api/wallet-requestPermissions.js

```diff
@@ -65,119 +68,125 @@ async function requestPermissionsImplementation(
   const [requestedPermissions] = params;
   delete requestedPermissions[Caip25EndowmentPermissionName];
 
-  const [_grantedPermissions] = await requestPermissionsForOrigin(
-    requestedPermissions,
-  );
-
-  // caveat values are frozen and must be cloned before modified
-  const grantedPermissions = { ..._grantedPermissions };
+  const legacyRequestedPermissions = pick(requestedPermissions, [
+    RestrictedMethods.eth_accounts,
+    PermissionNames.permittedChains,
+  ]);
+  delete requestedPermissions[RestrictedMethods.eth_accounts];
+  delete requestedPermissions[PermissionNames.permittedChains];
+
+  let legacyApproval;
+  if (Object.keys(legacyRequestedPermissions).length > 0) {
+    if (!legacyRequestedPermissions[RestrictedMethods.eth_accounts]) {
```

Comment:
🤔 not sure this case is possible. Its possible we want to throw an error in this case. `wallet_requestPermissions` will require a selection of at least one account and one network

Reply:
wait nvm this is just cleaning up the request before submitting it to the approval controller!